File: DSAinJava/BinaryTree/Binarytreerightsideview.java
/**
 * leetcode problem nu : 199
 * 
 * leetcode link(https://leetcode.com/problems/binary-tree-right-side-view/)
 */

 import java.util.*;

 class TreeNode {
     int val;
     TreeNode left;
     TreeNode right;
     TreeNode(int val) { this.val = val; }
 }
 
 public class Binarytreerightsideview {
     public List<Integer> rightSideView(TreeNode root) {
         List<Integer> result = new ArrayList<>();
         if (root == null)
             return result;
 
         Queue<TreeNode> queue = new LinkedList<>();
         queue.add(root);
 
         while (!queue.isEmpty()) {
             int n = queue.size();
             TreeNode rightNode = null;
 
             for (int i = 0; i < n; i++) {
                 rightNode = queue.poll();
 
                 if (rightNode.left != null)
                     queue.add(rightNode.left);
                 if (rightNode.right != null)
                     queue.add(rightNode.right);
             }
             result.add(rightNode.val);
         }
         return result;
     }
 
     public static void main(String[] args) {
         // Example usage:
         TreeNode root = new TreeNode(1);
         root.left = new TreeNode(2);
         root.right = new TreeNode(3);
         root.left.right = new TreeNode(5);
         root.right.right = new TreeNode(4);
 
         Binarytreerightsideview sol = new Binarytreerightsideview();
         List<Integer> rightView = sol.rightSideView(root);
         System.out.println(rightView);  // Output: [1, 3, 4]
     }
 }
 

File: DSAjavascript/BinarryTree/BinaryTreerightsideview.js
/**
 * leetcode problem nu : 199
 * 
 * leetcode link(https://leetcode.com/problems/binary-tree-right-side-view/)
 */
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number[]}
 */

function TreeNode(val, left, right) {
    this.val = (val === undefined ? 0 : val);
    this.left = (left === undefined ? null : left);
    this.right = (right === undefined ? null : right);
}

var rightSideView = function(root) {
    let result = [];
    
    if (!root) 
        return result;
    
    let queue = [];
    queue.push(root);
    
    while (queue.length > 0) {
        let n = queue.length;
        let rightNode = null;
        
        for (let i = 0; i < n; i++) {
            rightNode = queue.shift();
            
            if (rightNode.left !== null)
                queue.push(rightNode.left);
            
            if (rightNode.right !== null)
                queue.push(rightNode.right);
        }
        result.push(rightNode.val);
    }
    
    return result;
};

// Example usage
let root = new TreeNode(1);
root.left = new TreeNode(2);
root.right = new TreeNode(3);
root.left.right = new TreeNode(5);
root.right.right = new TreeNode(4);

console.log(rightSideView(root));  // Output: [1, 3, 4]